### PR TITLE
Findev/feat/adding njwt

### PR DESCRIPTION
**Pull request by GIANUNISINOS**
feats:

- Insert a user
- get a user by email
- generate token using njtw
- hashing the password to persist and compare
- receiving requires by JSON on body 

fix: 

- ajusts on connection

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,22 +1,114 @@
...
 });
 
+app.get('/api/user/all', async (req, res) => {
+  asyncConnection().then(async () => {
+      let uc= new UserController
+      res.send(await uc.getUsers())
+  })
+});
+
+/* ONLY FOR TESTS of token verify
```
Comment (joicecg) [inline src/index.ts]: Remove the comments.

File: src/index.ts [hunk 2]
```diff
@@ -1,22 +1,114 @@
...
+  "name":"Teste",
+  "password":"Gian",
+  "email":"gb@icloud.com",
+  "usertype":"EMPLOYEE"
+  }
+*/
+app.post('/api/user/insert', jsonParser, async (req, res) => {
+  try {
+      let createdUser = new User()
+      let uc= new UserController
```
Comment (joicecg) [inline src/index.ts]: Use a more meaningful name.

File: src/index.ts [hunk 3]
```diff
@@ -1,22 +1,114 @@
...
+ 
+      createdUser.NewUser(
+        req.body.name,
+        req.body.email,
+        uc.hashPassword(req.body.password),
+        req.body.usertype
+      )
+      asyncConnection().then(async () => {
+          uc.addUser(createdUser)
+          res.send("User Inserted on Database")
```
Comment (joicecg) [inline src/index.ts]: Change to `User successfully created`

File: src/index.ts [hunk 4]
```diff
@@ -1,22 +1,114 @@
...
+    })
+  } catch (error) {
+    res.send("User Doesn't Inserted on Database")
```
Comment (joicecg) [inline src/index.ts]: Change to `failed to create user`

**Review by joicecg — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (GIANUNISINOS) [reply src/index.ts]: removing all comments on index.ts

File: src/index.ts (hunk 2 above)
Comment (GIANUNISINOS) [reply src/index.ts]: changing to userController

File: src/index.ts (hunk 3 above)
Comment (GIANUNISINOS) [reply src/index.ts]: changing...

File: src/index.ts (hunk 4 above)
Comment (GIANUNISINOS) [reply src/index.ts]: ok

**Review by joicecg — APPROVED** [review]

**Pull request merged**